include/deal.II/lac/solver_idr.h

```diff
@@ -0,0 +1,481 @@
+} // namespace internal
+
+
+
+template <class VectorType>
+SolverIDR<VectorType>::SolverIDR(SolverControl &           cn,
+                                 VectorMemory<VectorType> &mem,
+                                 const AdditionalData &    data)
+  : SolverBase<VectorType>(cn, mem)
+  , additional_data(data)
+{}
+
+
+
+template <class VectorType>
+SolverIDR<VectorType>::SolverIDR(SolverControl &cn, const AdditionalData &data)
+  : SolverBase<VectorType>(cn)
+  , additional_data(data)
+{}
+
+
+
+template <class VectorType>
+void
+SolverIDR<VectorType>::print_vectors(const unsigned int,
+                                     const VectorType &,
+                                     const VectorType &,
+                                     const VectorType &) const
+{}
+
+
+
+template <class VectorType>
+template <typename MatrixType, typename PreconditionerType>
+void
+SolverIDR<VectorType>::solve(const MatrixType &        A,
+                             VectorType &              x,
+                             const VectorType &        b,
+                             const PreconditionerType &preconditioner)
+{
+  LogStream::Prefix prefix("IDR(s)");
+
+  SolverControl::State iteration_state = SolverControl::iterate;
+  unsigned int         step            = 0;
+
+  const unsigned int s = additional_data.s;
+
+  // Define temporary vectors which do not do not depend on s
+  typename VectorMemory<VectorType>::Pointer r_pointer(this->memory);
+  typename VectorMemory<VectorType>::Pointer v_pointer(this->memory);
+  typename VectorMemory<VectorType>::Pointer vhat_pointer(this->memory);
+  typename VectorMemory<VectorType>::Pointer uhat_pointer(this->memory);
+  typename VectorMemory<VectorType>::Pointer ghat_pointer(this->memory);
+
+  VectorType &r    = *r_pointer;
+  VectorType &v    = *v_pointer;
+  VectorType &vhat = *vhat_pointer;
+  VectorType &uhat = *uhat_pointer;
+  VectorType &ghat = *ghat_pointer;
+
+  r.reinit(x, true);
+  v.reinit(x, true);
+  vhat.reinit(x, true);
+  uhat.reinit(x, true);
+  ghat.reinit(x, true);
+
+  // Initial residual
+  A.vmult(r, x);
+  r.sadd(-1.0, 1.0, b);
+
+  // Check for convergent initial guess
+  double res      = r.l2_norm();
+  iteration_state = this->iteration_status(step, res, x);
+  if (iteration_state == SolverControl::success)
+    return;
+
+  // Initialize sets of vectors/matrices whose size dependent on s
+  internal::SolverIDRImplementation::TmpVectors<VectorType> G(s, this->memory);
+  internal::SolverIDRImplementation::TmpVectors<VectorType> U(s, this->memory);
+  internal::SolverIDRImplementation::TmpVectors<VectorType> Q(s, this->memory);
+  FullMatrix<double>                                        M(s, s);
+
+  // Random number generator for vector entries of
+  // Q (normal distribution, mean=0 sigma=1)
+  boost::mt19937               rng;
+  boost::normal_distribution<> nd(0.0, 1.0);
+  boost::variate_generator<boost::mt19937 &, boost::normal_distribution<>>
```

Comment:
We now use the C++11 facilities here instead: https://en.cppreference.com/w/cpp/numeric/random
There are already other places in the library that use `std::mt19937` that you can grep for.